src/sqlfluff/core/config.py

```diff
@@ -547,3 +547,11 @@ def process_inline_config(self, config_line: str):
         config_path = [elem.strip() for elem in config_line.split(":")]
         # Set the value
         self.set_value(config_path[:-1], config_path[-1])
+
+    def process_raw_file_for_config(self, raw_str: str):
+        """Process a full raw file for inline config and update self."""
+        # Scan the raw file for config commands.
+        for raw_line in raw_str.splitlines():
+            if raw_line.startswith("-- sqlfluff"):
```

Comment:
I didn't know about this!

Complete tangent here...

On Friday, I was talking to someone on Slack who wanted MySQL to support multiple kinds of delimiters between MySQL statements (apparently `;` is common, but people can change it). I suggested he use `OneOf()` to set up a fixed set of delimiters, but this makes me wonder about whether it'd be easy to change delimiters in the middle of a file.

Reply:
Oh I didn't know about that - do you have links to any docs about it? I use mysql quite a lot and hadn't come across it.

Reply:
Here's one: https://www.mysqltutorial.org/mysql-stored-procedure/mysql-delimiter/

Reply:
Oh wow. That is serious dark magic. I'm not sure how best to support those custom delimiters. We might have to have it as something that we knowingly don't support for a while.

Reply:
No worries -- he's working on a fixed set of typical delimiters, and may have a PR soon.